dotcom-rendering/src/web/browser/islands/doHydration.ts

```diff
@@ -62,11 +63,27 @@ export const doHydration = (
 			);
 		})
 		.catch((error) => {
-			if (name && error.message.includes(name)) {
+			if (error instanceof Error) {
+				if (
+					window.guardian.config.isDev &&
+					name &&
+					error.message.includes(name)
+				) {
+					// eslint-disable-next-line no-console -- We want to log this
+					console.error(
+						`Error importing '${name}'. (🚨 Components must live in the root of /components and follow the [MyComponent].importable.tsx naming convention 🚨)`,
+					);
+				}
+				window.guardian.modules.sentry.reportError(
+					error,
+					'islands-do-hydration',
+				);
+			} else {
+				// eslint-disable-next-line no-console -- We want to log this
 				console.error(
-					`🚨 Error importing ${name}. Components must live in the root of /components and follow the [MyComponent].importable.tsx naming convention 🚨`,
+					'[islands-do-hydration] Unknown error type:',
+					error,
 				);
```

Comment:
I think the reason is that `sentry.reportError` will only except the `Error` type, whereas this exception would be if somebody did something like `throw "I'm feeling angry"` - where `error` is now a string.

I believe sentry still picks up on all `console.error` so it will end up there anyway, but I'm not sure we can use `reportError` to achieve it

Reply:
<img width="556" alt="image" src="https://user-images.githubusercontent.com/9575458/180226363-1bddd1ba-a551-4e8a-b070-ea3f8aa482bf.png">